src/Common/src/System/Security/Cryptography/DSAOpenSsl.cs

```diff
@@ -283,6 +302,30 @@ public override bool VerifySignature(ReadOnlySpan<byte> hash, ReadOnlySpan<byte>
                 return Interop.Crypto.DsaVerify(key, hash, openSslFormat);
             }
 
+            private void ThrowIfDisposed()
+            {
+                if (_key == null)
+                {
+                    throw new ObjectDisposedException(
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                        nameof(DSA)
+#else
+                        nameof(DSAOpenSsl)
+#endif
+                    );
+                }
+            }
+
+            private SafeDsaHandle GetKey()
+            {
+                ThrowIfDisposed();
```

Comment:
Should SetKey also call ThrowIfDisposed?

Reply:
Looks like all the rest do as defensive measures, so added one here, too.